src/control/server/ctl_storage.go

```diff
@@ -212,6 +212,9 @@ func (cs *ControlService) scanAssignedBdevs(ctx context.Context, nsps []*ctl.Scm
 				continue
 			}
 
+			// NOTE DAOS-14223: This metadata size calculation won't necessarily match
+			//                  the meta blob size on SSD if --md-size is specified in
+			//                  pool create command.
```

Comment:
What are the ramifications of this?

Reply:
With MD-on-SSD, To compute the maximal storage usable on a NVMe disk to store DATA, you need to know the size that will be used for storing metadata, wall, etc on this NVMe device.
This part of the codes makes the assumption that the metadata size will be at most equal to the size of the VOS file stored into the SCM.  This size is then send to the engine to compute the size of the WAL associated to these VOS files.
With this new parameter, this assumption will not be any more valid and thus this part will need to be redesign.

Reply:
the purpose of this PR is to enable the value to be passed into the engine, but the first iteration only supports manual setting. future iterations will provide auto calculation of the value based on total size or percentage specifications in the create call